src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -639,7 +875,12 @@ private native void commissionDevice(
       long deviceControllerPtr,
       long deviceId,
       @Nullable byte[] csrNonce,
-      @Nullable NetworkCredentials networkCredentials);
+      @Nullable NetworkCredentials networkCredentials,
+      @Nullable DeviceAttestationDelegate delegate,
+      int failSafeExpiryTimeout);
+
+  private native void continueCommissioning(
+      long deviceControllerPtr, long devicePtr, boolean ignoreAttestationFailure);
```

Comment:
Do we use deviceControllerPtr to store the raw pointer?

Reply:
> Do we use deviceControllerPtr to store the raw pointer?

Do you mean to use the pointer from the DeviceAttestationFailureCallback or DeviceAttestationCompletionCallback callback as a parameter?